neurolib/utils/signal.py

```diff
@@ -0,0 +1,515 @@
+"""
+Base classes for representing signals.
+"""
+
+import logging
+from copy import deepcopy
+
+import numpy as np
+import xarray as xr
+from mne import set_log_level
+from mne.filter import filter_data, resample
+from scipy.signal import detrend, get_window, hilbert
+
+NC_EXT = ".nc"
+MNE_LOGGING_LEVEL = "WARNING"
+set_log_level(MNE_LOGGING_LEVEL)
+
+
+class Signal:
+    name = ""
+    label = ""
+    signal_type = ""
+    unit = ""
+    description = ""
+    _copy_attributes = [
+        "name",
+        "label",
+        "signal_type",
+        "unit",
+        "description",
+        "process_steps",
+    ]
+
+    @classmethod
+    def from_file(cls, filename):
+        """
+        Load signal from saved file.
+
+        :param filename: filename for the Signal
+        :type filename: str
+        """
+        if not filename.endswith(NC_EXT):
+            filename += NC_EXT
+        # load NC file
+        xarray = xr.load_dataarray(filename)
```

Comment:
Maybe we should adopt a unified way of saving arrays / signals / etc. This sounds interesting, if we adopt `xarray` throughout the project.